src/ripple/app/paths/PathRequest.cpp

```diff
@@ -690,10 +689,12 @@ Json::Value PathRequest::doUpdate(
 
     Json::Value& jvArray = (newStatus[jss::alternatives] = Json::arrayValue);
     if (! findPaths(cache, iLevel, jvArray))
+    {
+        bLastSuccess = false;
         newStatus = rpcError(rpcINTERNAL);
-
-    bLastSuccess = jvArray.size();
-    iLastLevel = iLevel;
+    }
+    else
+        bLastSuccess = jvArray.size() != 0;
 
```

Comment:
This fixes the issue but it's flakey: the `jvArray` still dangles we just no longer access it. Someone could trivially resurrect this bug by trying to simplify the code, and it's not immediately obvious code review would catch it.

Without changing `findPaths` to return the array, I find this to be more robust: 

```Json::Value array = Json::arrayValue;

if (findPaths (cache, iLevel, array))
{
    bLastSuccess = (array.size() == 0);
    newStatus[jss::alternatives] = std::move(array);
}
else
{
    bLastSuccess = false;
    newStatus = rpcError (rpcINTERNAL);
}```